### PR TITLE
Add release instructions

### PR DESCRIPTION
**Pull request by dgrisonnet**
## Type of change

_What type of changes does your code introduce to the kube-prometheus? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. Later this will be copied to the changelog file._

```release-note

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +105,12 @@ $ minikube addons disable metrics-server
 
 The following versions are supported and work as we test against these versions in their respective branches. But note that other versions might work!
 
+<!---
+**NOTE WHEN UPDATING THE MATRIX**: make sure to also apply the changes to the CI workflows.
+1. Update the [e2e-tests job](/.github/workflows/ci.yaml) to test HEAD against all compatible Kubernetes versions.
+2. Update the list of supported release branch in the [upgrade workflow](/.github/workflows/versions.yaml).
```
Comment (paulfantom) [inline README.md:111]: What do you think about creating `RELEASE.md` doc and put this info there?

Comment (dgrisonnet) [reply README.md:111]: sounds good

**Comment by dgrisonnet** [comment]
@paulfantom I updated this PR to add release instructions as per our previous discussion. PTAL

**Comment by dgrisonnet** [comment]
Added CHANGELOG update to the release process.

**Review by paulfantom — APPROVED** [review]
Awesome! Thanks!

**Pull request merged**